src/commands/serve.rs

```diff
@@ -72,7 +86,11 @@ fn serve_foreground(args: &ArgMatches) {
 
     info!("Starting HTTP server...");
 
-    let port = args.value_of("port").unwrap_or(DEFAULT_PORT);
+    let port = network::find_port(parse_default_port(args.value_of("port").unwrap_or(DEFAULT_PORT)));
+    if port.is_none() {
+        panic!("Unable to detect an available port, use --port with a value lower than 65535");
```

Comment:
is there something more stylish to exit with a formatted error message?

Reply:
Yes, there is, by using the `console::style()` function, but I'm not sure for now. Using panic is like throwing exceptions, it's perfectly fine for now since it does all three things that are needed: display an error message, stop the process, return a non-zero exit code 😄 

There's an example of usage in `proxy_server.rs` if you need :)

Reply:
I was just thinking that the output of panic is unclear, but then this case should rarely happen. Let's leave it for now, can be improved later on if people complain.